src/ext/serializer.c

```diff
@@ -0,0 +1,130 @@
+#include <php.h>
+#include <stdio.h>
+#include "mpack/mpack.h"
+
+static void ddtrace_zval_to_writer(mpack_writer_t *writer, zval *trace);
+
+#if PHP_VERSION_ID < 70000
+static void ddtrace_hash_table_to_writer(mpack_writer_t *writer, HashTable *ht) /* {{{ */
```

Comment:
similarliy to suggestion bellow:

the "ddtrace_" prefix is unnecessary, and at least to my ears `write_hash_table` sounds a bit nicer WRT intent of the function.